### PR TITLE
add a version label to search results

### PR DESCRIPTION
**Pull request by notwaldorf**
Looks like this. I don't think the search results are sorted by 2.0 first, which kinda sucks.

screen shot 2017-04-19 at 5 59 52

@keanulee @arthurevans

### CODE REVIEW

**Comment by arthurevans** [comment]
OK, so here's how you can filter for _only_ a single label: add the string 'more:<label>' to the query. So if the user query is "data binding" and we want to show 2.0, q should be: `q=data+binding+more:2.0`... (Also we should be running the query though `encodeURIComponent` or something similar).

File: app/search/index.html [hunk 1]
```diff
@@ -38,6 +38,26 @@
...
+
+      .search-result .version[data-version="1.0"] {
+        background-color: #37474f;
+      }
+
+      .search-result .version[data-version="2.0"] {
+        background-color: #1565c0;
+      }
+
+      .search-result .version[data-version="Blog"] {
```
Comment (keanulee) [inline app/search/index.html]: /s/version/label/, here and everwhere

Comment (notwaldorf) [inline app/search/index.html]: why? it's a version, no?

Comment (keanulee) [inline app/search/index.html]: "Blog" is not a version

Comment (notwaldorf) [inline app/search/index.html]: ok fair

**Comment by notwaldorf** [comment]
@arthurevans @keanu. Added a filter for results. By default:
screen shot 2017-04-25 at 10 34 59

Only for 1.0:
screen shot 2017-04-25 at 10 35 05

File: app/elements/pw-shell.html [hunk 2]
```diff
@@ -445,6 +448,7 @@
     },
 
     _handleSearchResponse: function(event) {
+      var i;  // the linter which may not be named has issues.
```
Comment (notwaldorf) [inline app/elements/pw-shell.html]: A story: I tried to make the comment be "// jshint has issues with scoping" and then `jshint` tried to interpret "has issues with scoping" as weird config parameters and anyway, `jshint` is Voldemort now, this is our darkest timeline.

**Comment by notwaldorf** [comment]
@arthurevans is this good to merge then?

File: app/elements/pw-shell.html [hunk 3]
```diff
@@ -458,10 +462,41 @@
...
-        var queryPrefix = '?q=' + this._searchParams.q + '&start=';
+        // Restricted search buttons. Ugh, so. we use the (undocumented) +more:version
+        // search parameter, which means we need to uhhh remove the +more
+        // keyword by hand if it already exists.
+        var matches = this._searchParams.q.match(/(.*) more:(1\.0|2\.0|Blog)/);
+        var searchParams;
+        var activeLabel;
+        if (matches && matches.length > 2) {
+          searchParams = matches[1];
+          activeLabel = matches[2]
```
Comment (arthurevans) [inline app/elements/pw-shell.html]: jshint is throwing a hissy about a missing semicolon here.

Comment (notwaldorf) [inline app/elements/pw-shell.html]: oshit good find, thanks (mine isn't, but i also don't update my jshint out of fear)

**Comment by arthurevans** [comment]
LGTM with one nit, when you select a link, it shows "Search results for test chai more:2.0"

Can we remove the "more:20" from the text?

**Comment by notwaldorf** [comment]
@arthurevans donzo

**Comment by arthurevans** [comment]
One more nitpick:

If I search for something that doesn't exist in a given category:

`/search/?q=chai+more:Blog`

The `more:` shows up on the "no results" page:

Search results for chai more:Blog
Your search — chai more:Blog — did not match any documents.

Also, ideally we'd include the filter links on this page so they can expand back to "all" or something.

**Comment by notwaldorf** [comment]
@arthurevans done. it wasn't the greatest solution -- that page needs refactoring so that the common code (title + filters) is just in one place rather than in two, but it would take a while and maybe we want to do it in a separate PR

**Comment by arthurevans** [comment]
Awesome! 🚀

**Pull request merged**